### PR TITLE
[BugFix] fix invalid nullable of materialized view

### LINKED ISSUE #21813: mv columns' nullable property is wrong
### Steps to reproduce the behavior (Required)
1. create lineorder_flat_mv with ssb test
2. the nullable properties of lineorder_flat_mv columns are all nullable

### Expected behavior (Required)
the nullable properties of lineorder_flat_mv columns should be based on the base tables and exprs
### Real behavior (Required)
the nullable properties of lineorder_flat_mv columns are all true
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by ABingHuang**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #21813 

## Problem Summary(Required) ：

Fix invalid nullable of materialized view. if the nullable is wrong,
the query performance may be affected after mv rewrite.
Fix it by get nullable info of column in Analyzer phase

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java [hunk 1]
```diff
@@ -312,11 +312,10 @@ public Scope visitTable(TableRelation node, Scope outerScope) {
                 Field field;
                 if (baseSchema.contains(column)) {
                     field = new Field(column.getName(), column.getType(), tableName,
-                            new SlotRef(tableName, column.getName(), column.getName()), true);
+                            new SlotRef(tableName, column.getName(), column.getName()), true, column.isAllowNull());
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java:315]: I think we could directly set the nullable property to SlotRef

Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java:315]: the expr is resolved from bottom to up. the path is: resolve table -> resolve SlotRef. After resolving table, we got scope and related fields. when resolving SlotRef, we get nullable from fields. so I should keep nullable info here

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java [hunk 2]
```diff
@@ -358,6 +360,9 @@ public Void visitSlot(SlotRef node, Scope scope) {
             ResolvedField resolvedField = scope.resolveField(node);
             node.setType(resolvedField.getField().getType());
             node.setTblName(resolvedField.getField().getRelationAlias());
+            // help to get nullable info in Analyzer phase
+            // now it is used in creating mv to decide nullable of fields
+            node.setNullable(resolvedField.getField().isNullable());
```
Comment (LiShuMing) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java:365]: is `resolvedField.getField().isNullable()`  always correct?

**Review by Astralidea — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java (hunk 2 above)
Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java:363]: now fields from base table, inner join, outer join are supported.
fields from expression use the default nullable deduction logica.

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java [hunk 3]
```diff
@@ -442,6 +441,24 @@ public Scope visitJoin(JoinRelation join, Scope parentScope) {
                 scope = new Scope(RelationId.of(join), leftScope.getRelationFields());
             } else if (join.getJoinOp().isRightSemiAntiJoin()) {
                 scope = new Scope(RelationId.of(join), rightScope.getRelationFields());
+            } else if (join.getJoinOp().isLeftOuterJoin()) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java:444]: what about full outer join?

Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java:444]: fixed

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java (hunk 2 above)
Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java:363]: yes

**Review by kangkaisen — DISMISSED** [review]

**Review by Astralidea — DISMISSED** [review]

**Review by packy92 — DISMISSED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 53 / 53 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/SlotRef.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|22|22|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/ColumnRewriter.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Field.java|20|20|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|1|1|100.00%|[]|

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**